include/seqan3/core/detail/transfer_type_modifier_onto.hpp

```diff
@@ -0,0 +1,70 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
+ * \brief Provides type traits seqan3::detail::transfer_type_modifier_onto.
+ */
+
+#pragma once
+
+#include <seqan3/std/type_traits>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Transfers the type modifier `&`, `&&` and `const` (and any combination) to the target type.
+ * \implements seqan3::transformation_trait
+ * \tparam source_t The type you wish to transfer the type modifier from.
+ * \tparam target_t The type you wish to transfer the type modifier to.
+ * \ingroup core
+ * If the target_t already has a type modifier, lets say `const`, it will keep that type modifier.
```

Comment:
```suggestion
 * If the target_t already has a type modifier, let's say `const`, it will keep that type modifier.
```
Instead of "let's say" you can use a more formal "e.g.".